modules/packages/UnorderedCopy.chpl

```diff
@@ -76,33 +82,32 @@
  */
 module UnorderedCopy {
   /*
-     Unordered copy. Only supported for identical numeric and bool types.
+     Unordered copy. Only supported for identical trivially copyable types.
    */
-  inline proc unorderedCopy(ref dst:numeric, const ref src:numeric): void {
-    unorderedCopyPrim(dst, src);
+  // Version to provide a clean signature for docs and to provide a clean error
+  // message instead of just "unresolved call". Last resort to avoid thwarting
+  // promotion for POD arrays.
+  pragma "last resort"
+  inline proc unorderedCopy(ref dst, src): void {
+    compilerError("unorderedCopy is only supported between identical trivially copyable types");
   }
 
   pragma "no doc"
-  inline proc unorderedCopy(ref dst:numeric, param src:numeric): void {
-    const refSrc = src;
-    unorderedCopyPrim(dst, refSrc);
-  }
-
-  inline proc unorderedCopy(ref dst:bool(?), const ref src:bool(?)): void {
+  inline proc unorderedCopy(ref dst:chpl_anyPOD, const ref src:chpl_anyPOD): void {
```

Comment:
If you wanted to enforce "identical" wouldn't you write

`ref dst:chpl_anyPOD, const ref src:dst.type`?

Reply:
I don't think that works like you're expecting it to. 


```chpl
proc myproc(a: chpl_anyPOD, b:a.type) {
  compilerWarning(a.type:string);
  compilerWarning(b.type:string);
}
myproc(1, 1.1);
```

```
f.chpl:5: warning: int(64)
f.chpl:5: warning: real(64)
```

(I'm guessing the .type resolved to chpl_anyPOD, not the instantiated type.

Reply:
Happens for `numeric` too. I wasn't sure if this was expected behavior or not (and I'm not sure I have a strong opinion about what the right behavior is.)

Reply:
Ah, this is issue #9310.